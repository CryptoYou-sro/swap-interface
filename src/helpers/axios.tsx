import { useCallback, useEffect, useState } from 'react';
import { Buffer } from 'buffer';
import {
	BASE_URL,
	BINANCE_DEV_URL,
	BINANCE_SCRIPT,
	BIZ_ENTRY_KEY,
	BINANCE_EXCHANGE_INFO,
	BINANCE_PRICE_TICKER
} from './index';
import axios from 'axios';
import destinationNetworks from '../data/destinationNetworks.json';

export enum STATUS_ENUM {
	NONCE = 'NONCE',
	AUTH = 'AUTH',
	PASS = 'PASS'
}

type LocalStorageProps = {
	isKyced: boolean;
	refresh: string;
	access: string;
	account: string;
};

export const apiCall = {
	getNonce: 'nonce?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'refresh'
};

export const getMetamaskMessage = (nonce: string): string =>
	`0x${Buffer.from('Please sign this one time nonce: ' + nonce, 'utf8').toString('hex')}`;

export const loadBinanceKycScript = (cb?: any) => {
	const existingId = document.getElementById('binance-kcy-script');

	if (!existingId) {
		const binanceSdkScript = document.createElement('script');
		binanceSdkScript.src = BINANCE_SCRIPT;
		binanceSdkScript.id = 'binance-kcy-script';
		document.body.appendChild(binanceSdkScript);

		binanceSdkScript.onload = () => {
			if (cb) cb();
		};
	}
	if (existingId && cb) cb();
};

export const makeBinanceKycCall = (authToken: string) => {
	// @ts-ignore
	const binanceKyc = new BinanceKyc({
		authToken,
		bizEntityKey: BIZ_ENTRY_KEY,
		apiHost: BINANCE_DEV_URL,
		onMessage: ({ typeCode }: any) => {
			if (typeCode === '102') {
				binanceKyc.switchVisible(true);
			}
		}
	});
};

const getStorageValue = (key: string) => {
	const defaultValue = {
		access: '',
		refresh: '',
		account: '',
		isKyced: false
	} as LocalStorageProps;

	if (typeof window !== 'undefined') {
		const saved = localStorage.getItem(key);

		return saved ? (JSON.parse(saved) as LocalStorageProps) : defaultValue;
	}
};

export const useLocalStorage = (key: string) => {
	const [value, setValue] = useState(() => {
		return getStorageValue(key);
	});

	useEffect(() => {
		localStorage.setItem(key, JSON.stringify(value));
	}, [key, value]);

	return [value, setValue];
};

export const getAuthTokensFromNonce = async (account: string, library: any) => {
	try {
		const res = await axios.request({
			url: `${BASE_URL}${apiCall.getNonce}${account}`
		});
		try {
			const msg = getMetamaskMessage(res.data.nonce);
			const signature = await library?.send('personal_sign', [account, msg]);
			try {
				const tokenRes = await axios.request({
					url: `${BASE_URL}${apiCall.auth}`,
					method: 'POST',
					data: { address: account, signature }
				});

				return tokenRes.data as { access: string; is_kyced: boolean; refresh: string }; // TODO: define Type
			} catch (err: any) {
				throw new Error(err);
			}
		} catch (err: any) {
			throw new Error(err);
		}
	} catch (err: any) {
		throw new Error(err);
	}
};

export const useKyc = (
	authToken: string
): { loading: boolean; error: any; kycStatus: string; kycToken: string } => {
	const [loading, setLoading] = useState(false);
	const [error, setError] = useState(null);
	const [kycStatus, setKycStatus] = useState('');
	const [kycToken, setKycToken] = useState('');

	const fetchData = useCallback(
		async (authToken: string): Promise<void> => {
			try {
				setLoading(true);
				const statusRes = await axios.request({
					// TODO: check typing and if kycStatus is from right place
					url: `${BASE_URL}${apiCall.kycStatus}`,
					headers: {
						Authorization: `Bearer ${authToken}`
					}
				});
				setKycStatus(statusRes.data.statusInfo.kycStatus);
			} catch (err: any) {
				setError(err);
			} finally {
				setLoading(false);
			}

			try {
				setLoading(true);
				const tokenRes = await axios.request({
					url: `${BASE_URL}${apiCall.kycToken}`,
					headers: {
						Authorization: `Bearer ${authToken}`
					}
				});
				setKycToken(tokenRes.data.token);
			} catch (err: any) {
				setError(err);
			} finally {
				setLoading(false);
			}
		},
		[authToken]
	);

	useEffect(() => {
		void fetchData(authToken);
	}, [fetchData]);

	return { loading, error, kycStatus, kycToken };
};

export const useBinanceApi = () => {
	const [allPrices, setAllPrices] = useState<{ symbol: string; price: string }[]>([]);
	const [allFilteredPrices, setAllFilteredPrices] = useState<{ symbol: string; price: string }[]>(
		[]
	);
	const [allPairs, setAllPairs] = useState([]);
	const [allFilteredPairs, setAllFilteredPairs] = useState([]);

	const getExchangeInfo = async () => {
		try {
			const res = await axios.request({ url: BINANCE_EXCHANGE_INFO });
			setAllPairs(res.data.symbols);
		} catch (e: any) {
			throw new Error(e);
		}
	};

	const getTickerData = async () => {
		try {
			const res = await axios.request({ url: BINANCE_PRICE_TICKER });
			setAllPrices(res.data);
		} catch (e: any) {
			throw new Error(e);
		}
	};

	const uniqueTokens: string[] = Object.keys(destinationNetworks).reduce(
		(tokens: string[], network: string) => {
			// @ts-ignore
			const networkTokens = Object.keys(destinationNetworks?.[network]?.['tokens']);

			const allTokens = [...tokens, networkTokens];

			return [...new Set(allTokens.flat(1))];
		},
		['GLMR']
	);

	const isSymbol = (symbol: string): boolean => {
		let k = 0;
		for (let i = 0; i < uniqueTokens.length - 1; i++) {
			k++;
			for (let j = k; j < uniqueTokens.length; j++) {
				if (
					uniqueTokens[i] + uniqueTokens[j] === symbol ||
					uniqueTokens[j] + uniqueTokens[i] === symbol
				) {
					return true;
				}
			}
		}

		return false;
	};

	useEffect(() => {
		void getExchangeInfo();
		void getTickerData();
	}, []);

	useEffect(() => {
		if (allPairs) {
			const filteredPairs = allPairs.filter((pair: any) => isSymbol(pair.symbol));
			setAllFilteredPairs(filteredPairs);
		}

		if (allPrices) {
			const filteredPrices = allPrices.filter((price: any) => isSymbol(price.symbol));
			setAllFilteredPrices(filteredPrices);
		}
	}, [allPairs, allPrices]);

	return { allFilteredPairs, allFilteredPrices };
};
