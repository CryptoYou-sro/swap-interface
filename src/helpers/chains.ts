import { Chain } from '@wagmi/core';
 
export const moonbeam = {
    id: 1284,
    name: 'Moonbeam',
    network: 'moonbeam',
    nativeCurrency: {
        decimals: 18,
        name: 'GLMR',
        symbol: 'GLMR',
    },
    rpcUrls: {
        public: {
            http: ['https://rpc.api.moonbeam.network'],
            webSocket: ['wss://wss.api.moonbeam.network'],
        },
        default: {
            http: ['https://rpc.api.moonbeam.network'],
            webSocket: ['wss://wss.api.moonbeam.network'],
        },
    },
    blockExplorers: {
        default: {
            name: 'Moonscan',
            url: 'https://moonscan.io',
        },
        etherscan: {
            name: 'Moonscan',
            url: 'https://moonscan.io',
        },
    },
    contracts: {
        multicall3: {
            address: '0xcA11bde05977b3631167028862bE2a173976CA11',
            blockCreated: 609002,
        },
    },
    testnet: false
} as const satisfies Chain;

export const bsc = {
    id: 56,
    name: 'Binance Smart Chain',
    network: 'bsc',
    nativeCurrency: {
        decimals: 18,
        name: 'BNB',
        symbol: 'BNB',
    },
    rpcUrls: {
        public: {
            http: ['https://bsc.publicnode.com'],
        },
        default: {
            http: ['https://bsc.publicnode.com'],
        },
    },
    blockExplorers: {
        default: {
            name: 'BscScan',
            url: 'https://bscscan.com',
        },
        etherscan: {
            name: 'BscScan',
            url: 'https://bscscan.com',
        },
    },
    contracts: {
        multicall3: {
            address: '0xca11bde05977b3631167028862be2a173976ca11',
            blockCreated: 15921452,
        },
    },
    testnet: false
} as const satisfies Chain;
