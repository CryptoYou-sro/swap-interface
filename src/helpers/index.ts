export * from './functions';
export * from './context';
export * from './constants';
export * from './auth';
export * from './graph';
export * from './storage';
export * from './types';
