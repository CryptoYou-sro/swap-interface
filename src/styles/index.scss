@import "./fonts/font-face.scss";
@import "./variables.scss";
