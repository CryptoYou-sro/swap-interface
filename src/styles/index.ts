export * from './color';
export * from './spacing';
export * from './font';
export * from './mediaQuery';
export * from './types';
