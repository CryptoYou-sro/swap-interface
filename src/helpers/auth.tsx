import { Buffer } from 'buffer';
import { BASE_URL, BINANCE_PROD_URL, BINANCE_SCRIPT, BIZ_ENTRY_KEY } from '../helpers';
import axios from 'axios';
import type { ApiAuthType } from '../helpers';

export enum STATUS_ENUM {
	NONCE = 'NONCE',
	AUTH = 'AUTH',
	PASS = 'PASS'
}

export const routes = {
	getNonce: 'nonce?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'auth/refresh'
};

export const getMetamaskMessage = (nonce: string): string =>
	`0x${Buffer.from('Please sign this one time nonce: ' + nonce, 'utf8').toString('hex')}`;

export const loadBinanceKycScript = (cb?: any) => {
	const existingId = document.getElementById('binance-kcy-script');

	if (!existingId) {
		const binanceSdkScript = document.createElement('script');
		binanceSdkScript.src = BINANCE_SCRIPT;
		binanceSdkScript.id = 'binance-kcy-script';
		document.body.appendChild(binanceSdkScript);

		binanceSdkScript.onload = () => {
			if (cb) cb();
		};
	}
	if (existingId && cb) cb();
};

export const makeBinanceKycCall = (authToken: string) => {
	// @ts-ignore
	const binanceKyc = new BinanceKyc({
		authToken,
		bizEntityKey: BIZ_ENTRY_KEY,
		apiHost: BINANCE_PROD_URL,
		onMessage: ({ typeCode }: any) => {
			if (typeCode === '102') {
				binanceKyc.switchVisible(true);
			}
		}
	});
};

export const getAuthTokensFromNonce = async (account: string, library: any) => {
	try {
		const res = await axios.request({
			url: `${BASE_URL}${routes.getNonce}${account}`
		});
		try {
			const msg = getMetamaskMessage(res.data.nonce);
			const signature = await library?.send('personal_sign', [account, msg]);
			try {
				const tokenRes = await axios.request({
					url: `${BASE_URL}${routes.auth}`,
					method: 'POST',
					data: { address: account, signature }
				});

				return tokenRes.data as ApiAuthType;
			} catch (err: any) {
				throw new Error(err);
			}
		} catch (err: any) {
			throw new Error(err);
		}
	} catch (err: any) {
		throw new Error(err);
	}
};
