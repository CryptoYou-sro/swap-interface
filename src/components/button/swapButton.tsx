import { forwardRef, useImperativeHandle } from 'react';
import { ERC20Interface, useContractFunction, useEthers, useSendTransaction } from '@usedapp/core';
import styled from 'styled-components';
import CONTRACT_DATA from '../../data/YandaExtendedProtocol.json';
import { useContractFunction, useEthers, useSendTransaction } from '@usedapp/core';
import { utils } from 'ethers';
import { Button } from '..';
import { Contract } from '@ethersproject/contracts';
import {
	CONTRACT_ADDRESSES,
	isNetworkSelected,
	isTokenSelected,
	makeId,
	SERVICE_ADDRESS,
	useStore,
	START_TOKEN
} from '../../helpers';
import type { ContractAdress } from '../../helpers';
import CONTRACT_DATA from '../../data/YandaMultitokenProtocolV1.json';
import sourceNetworks from '../../data/sourceNetworks.json';
import styled from 'styled-components';
import { spacing } from '../../styles';

const ButtonWrapper = styled.div`
	margin-top: ${spacing[28]};
`;

type Props = {
	validInputs: boolean;
	amount: string;
	onClick: () => void;
};

export const SwapButton = forwardRef(({ validInputs, amount, onClick }: Props, ref) => {
	const {
		state: {
			destinationNetwork,
			destinationToken,
			destinationAddress,
			destinationMemo,
			isUserVerified,
			destinationAmount
		}
	} = useStore();
	const isDisabled =
		!validInputs ||
		!isNetworkSelected(destinationNetwork) ||
		!isTokenSelected(destinationToken) ||
		!isUserVerified ||
		+destinationAmount < 0;

	const { account, chainId, library: web3Provider } = useEthers();
	const startTokenData =
		// @ts-ignore
		sourceNetworks[chainId.toString()]?.tokens[START_TOKEN];
	const tokenContract = startTokenData.contractAddr && new Contract(startTokenData.contractAddr, ERC20Interface, web3Provider);

	const protocolAddress = CONTRACT_ADDRESSES?.[chainId as ContractAdress] || '';
	const protocolInterface = new utils.Interface(CONTRACT_DATA.abi);
	const protocol = new Contract(protocolAddress, protocolInterface, web3Provider);
	if (web3Provider) {
		protocol.connect(web3Provider.getSigner());
		if (tokenContract) {
			tokenContract.connect(web3Provider.getSigner());
		}
	}

	const { send: sendTokenApprove } = useContractFunction(
		// @ts-ignore
		tokenContract,
		'approve',
		{ transactionName: 'Approve token to be used for Swap' }
	);
	const { send: sendCreateProcess } = useContractFunction(
		// @ts-ignore
		protocol,
		'createProcess(address,bytes32,string)',
		{ transactionName: 'Request Swap' }
	);
	const { send: sendTokenCreateProcess } = useContractFunction(
		// @ts-ignore
		protocol,
		'createProcess(address,address,bytes32,string)',
		{ transactionName: 'Request Swap' }
	);
	const { send: sendDeposit } = useContractFunction(
		// @ts-ignore
		protocol,
		'deposit',
		{ transactionName: 'deposit' }
	);
	const { sendTransaction } = useSendTransaction({
		transactionName: 'Deposit'
	});

	useImperativeHandle(ref, () => ({
		async onSubmit() {
			const productId = utils.id(makeId(32));

			const namedValues = {
				scoin: 'ETH',
				samt: utils.parseEther(amount).toString(),
				fcoin: destinationToken,
				net: destinationNetwork,
				daddr: destinationAddress,
				tag: destinationMemo
			};

			const shortNamedValues = JSON.stringify(namedValues);

			if(startTokenData.isNative) {
				await sendCreateProcess(SERVICE_ADDRESS, productId, shortNamedValues);
			} else {
				console.log('Calling sendCreateProcess with the contractAddr', startTokenData.contractAddr);
				await sendTokenCreateProcess(startTokenData.contractAddr, SERVICE_ADDRESS, productId, shortNamedValues);
			}
			const filter = protocol.filters.CostResponse(account, SERVICE_ADDRESS, productId);
			protocol.on(filter, (customer, service, productId, cost) => {
				console.log('Oracle deposit estimation:', utils.formatEther(cost));
				if(startTokenData.isNative) {
					void sendTransaction({ to: protocolAddress, value: cost });
				} else {
					sendTokenApprove(protocolAddress, cost)
						.then(result => {
							console.log('Approved ', utils.formatEther(cost), ' tokens of "', protocolAddress, '" contract.', result);

							void sendDeposit(cost);
						})
						.catch(error => {
							console.log('Error in sending approve', error);
						});
				}
			});
		}
	}));

	return (
		<ButtonWrapper>
			<Button disabled={isDisabled} color="default" onClick={onClick}>
				SWAP
			</Button>
		</ButtonWrapper>
	);
});
