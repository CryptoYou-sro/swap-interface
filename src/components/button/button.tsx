import type { ReactNode } from 'react';
import { pxToRem } from '../../styles';
import styled, { css } from 'styled-components';
import moonbeam from '../../assets/moonbeam.svg';
import metamask from '../../assets/metamask.svg';
import { defaultBorderRadius, isLightTheme, useStore } from '../../helpers';

const icons = {
	moonbeam,
	metamask
};

export type ColorType = 'default' | 'error' | 'warning' | 'icon' | 'success';

interface CommonProps {
	disabled?: boolean;
	children?: ReactNode;
	onClick: () => void;
}

interface PrimaryProps {
	variant?: 'primary';
	color: 'default' | 'transparent';
	icon?: never;
}

interface PureProps {
	variant?: 'pure';
	color?: never;
	icon?: never;
}

interface IconOnlyProps {
	icon?: 'moonbeam' | 'metamask';
	color?: never;
}

interface ColorOnlyProps {
	color?: ColorType;
	icon?: never;
}

type ColorIconProps = IconOnlyProps | ColorOnlyProps;

type SecondaryProps = {
	variant: 'secondary';
} & ColorIconProps;

type IndividualProps = PrimaryProps | SecondaryProps | PureProps;
type Props = IndividualProps & CommonProps;

const StyledButton = styled.button(
	({ variant = 'primary', color = 'default', disabled = false, icon }: Props) => {
		const isPrimary = variant === 'primary';
		const isSecondary = variant === 'secondary';
		const isPure = variant === 'pure';
		const setColor = icon ? 'icon' : color;
		const isColorDefault = setColor === 'default';
		const isPrimaryTransparent = variant === 'primary' && color === 'transparent';
		const isPrimaryDisabled = variant === 'primary' && disabled;
		const isSecondaryDefault = isSecondary && setColor === 'default';
		const {
			state: { theme }
		} = useStore();

		return css`
			display: ${icon ? 'inline-flex' : 'inline-block'};
			align-items: center;
			justify-content: space-between;
			max-width: ${isPrimary ? pxToRem(428) : pxToRem(160)};
			width: 100%;
			cursor: ${disabled ? 'not-allowed' : 'pointer'};
			font-size: ${isPrimary ? pxToRem(16) : pxToRem(14)};
			min-height: ${isPrimary ? pxToRem(57) : pxToRem(35)};
			padding: ${pxToRem(4)} ${pxToRem(12)};
			color: ${isPure
				? theme.pure
				: isSecondaryDefault || isPrimaryTransparent
				? theme.button.default
				: '#FFF'};
			background-color: ${isPure || isSecondaryDefault || isPrimaryTransparent
				? theme.button.transparent
				: disabled
				? theme.button.disabled
				: theme.button[setColor]};
			border: 1px solid
				${isPrimaryDisabled
					? theme.button.disabled
					: isSecondaryDefault || isPrimaryTransparent
					? theme.button.default
					: isPure || isColorDefault
					? theme.button.transparent
					: '#FFF'};
			border-radius: ${defaultBorderRadius};
			transition: all 0.2s ease-in-out;
			margin: ${isSecondaryDefault && '1px'};
			cursor: ${disabled && 'not-allowed'};

			&:hover {
				opacity: ${!isSecondaryDefault && '0.8'};
				box-shadow: ${isSecondaryDefault && `0 0 0 1px ${theme.button.default}`};
			}

			&:focus-visible {
				outline-offset: 2px;
				outline: 1px solid
					${isPrimary
						? theme.button.default
						: isPure
						? theme.pure
						: isLightTheme(theme)
						? theme.button[setColor]
						: '#FFF'};
			}

			&:active {
				outline: none;
			}
		`;
	}
);

export const Button = ({
	children,
	variant = 'primary',
	color = 'default',
	disabled = false,
	icon,
	onClick
}: Props) => {
	return (
		// @ts-ignore
		<StyledButton icon={icon} color={color} variant={variant} disabled={disabled} onClick={onClick}>
			{icon && <img src={icons?.[icon]} alt={icon} />}
			{children}
		</StyledButton>
	);
};
