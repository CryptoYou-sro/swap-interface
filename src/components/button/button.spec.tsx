import 'jest-styled-components';
import { render, fireEvent, screen, findByText, getByRole } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { AuthProvider } from '../../helpers';
import { lightTheme, darkTheme } from '../../styles';
import { Button } from './button';
import { pxToRem } from '../../styles';

describe('Button Component', () => {
	it('should render a default button', () => {

		const { getByText, getByRole } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')}>Primary default Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Primary default Button/)).toHaveStyle(
			`background-color: ${lightTheme.button.default}; color: #FFF; border: 1px solid transparent; max-width: ${pxToRem(428)}; cursor: pointer;`
		);

		expect(getByText(/Primary default Button/)).toMatchSnapshot();

		const btn = getByText(/Primary default Button/)
		userEvent.hover(btn);
		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});

	it('should render a secondary default button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant={'secondary'}>Secondary default Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary default Button/)).toHaveStyle(
			`background-color: transparent; color: ${lightTheme.button.default}; border: 1px solid ${lightTheme.button.default}; max-width: ${pxToRem(160)}`
		);

		expect(getByText(/Secondary default Button/)).toMatchSnapshot();

		const btn = getByText(/Secondary default Button/)
		userEvent.hover(btn);
		expect(btn).toHaveStyleRule("box-shadow", `0 0 0 1px ${lightTheme.button.default}`, {
			modifier: ':hover',
		});

	});
	it('should render a secondary icon button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant="secondary" icon="moonbeam">Secondary icon Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary icon Button/)).toHaveStyle(
			`background-color: ${lightTheme.button.icon}; color: #FFF; border: 1px solid #FFF; max-width: ${pxToRem(160)}`
		);

		expect(getByText(/Secondary icon Button/)).toMatchSnapshot();

		const btn = getByText(/Secondary icon Button/)
		userEvent.hover(btn);

		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});

	it('should render a secondary warning button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant={'secondary'} color={'warning'}>Secondary warning Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary warning Button/)).toHaveStyle(
			`background-color: ${lightTheme.button.warning}; color: #FFF; border: 1px solid #FFF; max-width: ${pxToRem(160)}`
		);

		expect(getByText(/Secondary warning Button/)).toMatchSnapshot();

		const btn = getByText(/Secondary warning Button/)
		userEvent.hover(btn);

		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});

	it('should render a secondary error button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant={'secondary'} color={'error'}>Secondary error Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary error Button/)).toHaveStyle(
			`background-color: ${lightTheme.button.error}; color: #FFF; border: 1px solid #FFF; max-width: ${pxToRem(160)}`
		);

		expect(getByText(/Secondary error Button/)).toMatchSnapshot();

		const btn = getByText(/Secondary error Button/)
		userEvent.hover(btn);

		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});

	it('should render a primary disabled button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant={'primary'} disabled>Secondary primary disabled Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary primary disabled Button/)).toHaveStyle(
			`background-color: ${lightTheme.button.disabled}; color: #FFF; border: 1px solid transparent; max-width: ${pxToRem(428)}`
		);

		expect(getByText(/Secondary primary disabled Button/)).toMatchSnapshot();


		const btn = getByText(/Secondary primary disabled Button/)
		userEvent.hover(btn);

		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});

	it('should render a pure button', () => {
		const { getByText } = render(
			<AuthProvider>
				<Button onClick={() => console.log('test')} variant={'pure'}>Secondary pure Button</Button>
			</AuthProvider>
		);
		expect(getByText(/Secondary pure Button/)).toHaveStyle(
			`background-color: transparent; color: #FFF; border: 1px solid transparent; max-width: ${pxToRem(160)}`
		);

		expect(getByText(/Secondary pure Button/)).toMatchSnapshot();


		const btn = getByText(/Secondary pure Button/)
		userEvent.hover(btn);

		expect(btn).toHaveStyleRule("opacity", "0.8", {
			modifier: ':hover',
		});
	});
});
