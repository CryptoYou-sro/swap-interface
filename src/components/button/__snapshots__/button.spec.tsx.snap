// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a default button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 28.125rem;
  width: 100%;
  cursor: pointer;
  font-size: 1rem;
  min-height: 3.5625rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #00A8E8;
  border: 1px solid transparent;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #00A8E8;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
>
  Primary default Button
</button>
`;

exports[`Button should render a primary disabled button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 28.125rem;
  width: 100%;
  cursor: not-allowed;
  font-size: 1rem;
  min-height: 3.5625rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #6D6D6D;
  border: 1px solid #6D6D6D;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: not-allowed;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #00A8E8;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
  disabled=""
>
  Primary disabled Button
</button>
`;

exports[`Button should render a primary disabled button 2`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 28.125rem;
  width: 100%;
  cursor: not-allowed;
  font-size: 1rem;
  min-height: 3.5625rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #6D6D6D;
  border: 1px solid #6D6D6D;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: not-allowed;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #00A8E8;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
  disabled=""
>
  Primary disabled Button
</button>
`;

exports[`Button should render a pure button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 10rem;
  width: 100%;
  cursor: pointer;
  font-size: 0.875rem;
  min-height: 2.1875rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #FFF;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
>
  Secondary pure Button
</button>
`;

exports[`Button should render a secondary default button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 10rem;
  width: 100%;
  cursor: pointer;
  font-size: 0.875rem;
  min-height: 2.1875rem;
  padding: 0.25rem 0.75rem;
  color: #00A8E8;
  background-color: transparent;
  border: 1px solid #00A8E8;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  margin: 1px;
}

.c0:hover {
  box-shadow: 0 0 0 1px #00A8E8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #FFF;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
>
  Secondary default Button
</button>
`;

exports[`Button should render a secondary error button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 10rem;
  width: 100%;
  cursor: pointer;
  font-size: 0.875rem;
  min-height: 2.1875rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #DE3434;
  border: 1px solid #FFF;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #FFF;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="error"
>
  Secondary error Button
</button>
`;

exports[`Button should render a secondary icon button 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 10rem;
  width: 100%;
  cursor: pointer;
  font-size: 0.875rem;
  min-height: 2.1875rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #172631;
  border: 1px solid #FFF;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #FFF;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="default"
>
  <img
    alt="moonbeam"
    src="moonbeam.svg"
  />
  Secondary icon Button
</button>
`;

exports[`Button should render a secondary warning button 1`] = `
.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 10rem;
  width: 100%;
  cursor: pointer;
  font-size: 0.875rem;
  min-height: 2.1875rem;
  padding: 0.25rem 0.75rem;
  color: #FFF;
  background-color: #FD862F;
  border: 1px solid #FFF;
  border-radius: 0.375rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0:focus-visible {
  outline-offset: 2px;
  outline: 1px solid #FFF;
}

.c0:active {
  outline: none;
}

<button
  class="c0"
  color="warning"
>
  Secondary warning Button
</button>
`;
